storage/wdpost_sched.go

```diff
@@ -17,10 +18,38 @@ import (
 	"github.com/filecoin-project/lotus/build"
 	"github.com/filecoin-project/lotus/chain/store"
 	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/journal"
 )
 
 const StartConfidence = 4 // TODO: config
 
+type WindowPoStEvt struct {
+	State    string
+	Deadline *miner.DeadlineInfo
+	Height   abi.ChainEpoch
+	TipSet   []cid.Cid
+	Error    error `json:",omitempty"`
+
+	Proofs     *WindowPoStEvt_Proofs     `json:",omitempty"`
+	Recoveries *WindowPoStEvt_Recoveries `json:",omitempty"`
+	Faults     *WindowPoStEvt_Faults     `json:",omitempty"`
```

Comment:
These could actually be splintered into separate event types.

Reply:
Probably worth doing, window post journal will likely be the one of the most looked at

Reply:
Makes sense, will do.

Reply:
Done.